src/PIL/ImageGrab.py

```diff
@@ -140,7 +140,14 @@ def grabclipboard():
                 return BmpImagePlugin.DibImageFile(data)
         return None
     else:
-        if shutil.which("wl-paste"):
+        if os.getenv("WAYLAND_DISPLAY"):
+            sessiontype = "wayland"
+        elif os.getenv("DISPLAY"):
+            sessiontype = "x11"
+        else:  # Session type check failed
+            sessiontype = None
+
+        if shutil.which("wl-paste") and sessiontype in ["wayland", None]:
```

Comment:
```suggestion
        if os.getenv("WAYLAND_DISPLAY"):
            session_type = "wayland"
        elif os.getenv("DISPLAY"):
            session_type = "x11"
        else:  # Session type check failed
            session_type = None

        if shutil.which("wl-paste") and session_type in ["wayland", None]:
```
https://peps.python.org/pep-0008/#function-and-variable-names
> Function names should be lowercase, with words separated by underscores as necessary to improve readability.
>
> Variable names follow the same convention as function names.